### PR TITLE
Changed docs deployment action

### PR DESCRIPTION
**Pull request by JAHMCR**
Changed docs deployment action to use GitHub verified action

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
       - 'v*'
 
 jobs:
-  Documentation:
+  build:
```
Comment (gegles) [inline .github/workflows/docs.yml:9]: Why change the name of this action? `Documentation` seemed more appropriate...

Comment (JAHMCR) [reply .github/workflows/docs.yml:9]: Sure, I can revert that change, I just followed the recommended names from the official doc [page](https://github.com/actions/deploy-pages)

Comment (gegles) [reply .github/workflows/docs.yml]: I see. You are probably right then... but, maybe then let's refactor all the workflows appropriately in the future (once we successfully release). thx!

**Review by gegles — APPROVED** [review]
LGTM, let's try...

**Pull request merged**